game-core/src/test/java/games/strategy/engine/xml/TestGameLoader.java

```diff
@@ -53,7 +55,7 @@ public void startGame(final IGame game, final Set<IGamePlayer> players,
   public void shutDown() {}
 
   @Override
-  public UnitFactory getUnitFactory() {
-    return Unit::new;
+  public Unit createUnit(final UnitType type, final PlayerID owner, final GameData data) {
+    return new Unit(type, owner, data);
```

Comment:
Agreed.  Because the serialization stream stores the complete class hierarchy, the best we can probably do to maintain compatibility is make `Unit` abstract and try to move as many method implementations as we can to `TripleAUnit`.  Then, upon the next incompatible release, merge the two back together as `Unit`.